src/shared/util/logic.ts

```diff
@@ -81,13 +95,13 @@ function groupLogicUnitsByField(form) {
  * @param {Object} form Form object
  * @returns {Array} Array of conditions to prevent submission
  */
-function getPreventSubmitConditions(form) {
+function getPreventSubmitConditions(form: IForm): IPreventSubmitLogicSchema[] {
   const formFieldIds = new Set(
     form.form_fields.map((field) => String(field._id)),
   )
   return form.form_logics.filter((formLogic) => {
     return (
-      formLogic.logicType === LOGIC_TYPES.preventSubmit &&
+      !isShowFieldsLogic(formLogic) &&
```

Comment:
Prefer a `isPreventSubmitLogic` to check whether the logic is `LogicType.PreventSubmit`, since we may add even more logic types in the future and may result in weird behavior.